### PR TITLE
APK generation for Moh Mali CHW

### PR DESCRIPTION
**Pull request by ernestoteo**
[#220](https://github.com/medic/config-moh-mali-chw/issues/220)

### CODE REVIEW

**Comment by ernestoteo** [comment]
@latin-panda , We created the Moh Mali CHW flavor, based on  [v1.0.0](https://github.com/medic/cht-android/tree/v1.0.0).  Should the merge be on [v1.0.x](https://github.com/medic/cht-android/tree/v1.0.x) ?

**Comment by latin-panda** [comment]
Hi @ernestoteo 

Thanks for your question. Yes, according with [the docs](https://docs.communityhealthtoolkit.org/core/guides/android/releasing/#new-flavor-release), once @binokaryg approves this PR, then:
1. `squash and merge` the PR into the release branch `v1.0.x`.
2. Tag the commit with the same version as the last one and add a number suffix separated with a dash character (-), for example: `v1.0.0-1`
3. That will create a new [release](https://github.com/medic/cht-android/releases) with the APKs. 
4. Make sure this commit is backported (cherry pick) to master branch.

cc: @garethbowen

**Comment by binokaryg** [comment]
@latin-panda thanks for the information. This can't be merged yet because we are still waiting on the logo and the app name from the partner. Also, the secrets need to be added as GitHub secrets before merging, right? Who can add them?

**Comment by latin-panda** [comment]
Hi @binokaryg, I think Derick or Marina can help you with that, usually managers have access to add secrets in Github. Product devs don't have access to that or any other settings in Github :)

(More about how to [generate secrets](https://docs.communityhealthtoolkit.org/core/guides/android/branding/#3-generate-a-new-keystore))

File: build.gradle [hunk 1]
```diff
@@ -115,7 +115,7 @@ android {
     buildConfigField "String", "TEST_USERNAME", "\"${System.env.ANDROID_TEST_USERNAME}\""
     buildConfigField "String", "TEST_PASSWORD", "\"${System.env.ANDROID_TEST_PASSWORD}\""
 
-    buildConfigField "boolean", "IS_TRAINING_APP", 'false'
+    buildConfigField "boolean", "IS_TRAINING_APP", 'true'
```
Comment (binokaryg) [inline build.gradle:118]: Should we make this `false`?

**Review by binokaryg — CHANGES_REQUESTED** [review]
I think it should not be a training app.

File: build.gradle (hunk 1 above)
Comment (ernestoteo) [reply build.gradle:118]: It should be false

**Comment by ernestoteo** [comment]
@derickl, can you please add the key store secrets of Moh Niger CHW, you can get them from 1Password.

File: build.gradle [hunk 2]
```diff
@@ -212,7 +212,7 @@ android {
     medicmobilegamma_training {
       dimension = 'brand'
       applicationId = 'org.medicmobile.webapp.mobile.medicmobilegamma_training'
-      buildConfigField "boolean", "IS_TRAINING_APP", 'true'
+      buildConfigField "boolean", "IS_TRAINING_APP", 'false'
```
Comment (latin-panda) [inline build.gradle:215]: I think this branch has changes that doesn't belong to `1.0.x`.

File: build.gradle (hunk 1 above)
Comment (latin-panda) [inline build.gradle:118]: same here

File: src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -19,11 +19,12 @@
   <uses-permission android:name="com.android.launcher.permission.INSTALL_SHORTCUT"/>
   <uses-permission android:name="com.android.launcher.permission.UNINSTALL_SHORTCUT"/>
 
-  <!-- READ_EXTERNAL_STORAGE required if users want to include photos from their phone  -->
+  <!-- READ_EXTERNAL_STORAGE is required if users want to include photos from their phone  -->
```
Comment (latin-panda) [inline src/main/AndroidManifest.xml:22]: same with this file.

**Review by latin-panda — CHANGES_REQUESTED** [review]
Hi @ernestoteo, looks like your branch has changes from `master` or something like that and those don't belong to `v1.0.x` branch, I've flagged the places below. This PR should contain only additions related to the new flavor MoH Mali CHW. Can you please have a look? Thanks!  
CC: @binokaryg

**Comment by latin-panda** [comment]
Because anyways you'll need the bug fixes from the patch release for Moh Mali, I think.

File: build.gradle (hunk 1 above)
Comment (binokaryg) [reply build.gradle:118]: It's still `true`.

File: build.gradle (hunk 2 above)
Comment (binokaryg) [inline build.gradle:215]: This is `true` in `v1.0.x`.

Comment (ernestoteo) [reply build.gradle:215]: So can we keep it `true` in this commit ?

**Comment by ernestoteo** [comment]
@binokaryg, @latin-panda I am getting this error [500 from server: Internal Server Error](https://github.com/medic/cht-android/runs/6979535364?check_suite_focus=true#step:8:137)

**Comment by latin-panda** [comment]
I've updated the based branch with the released patch (v1.0.1), MoH Mali will have the latest bug fixes. Now PR should only show the new additions :) 

@ernestoteo I think the third party library server was down, it had happened before. Let's see if this time the checks pass green

**Review by binokaryg — APPROVED** [review]
Looks good. Thanks for making changes.

**Pull request merged**